auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -471,23 +475,247 @@ public Builder(@SupportedProvider @NonNull String providerId) {
              * <a href="https://apps.twitter.com/">Twitter developer console</a>.
              */
             @NonNull
+            @Deprecated
             public Builder setPermissions(@Nullable List<String> permissions) {
-                mScopes = permissions;
+                if (permissions == null) {
+                    mParams.clear();
+                    return this;
+                }
+
+                Bundle params;
+                if (mProviderId.equals(GOOGLE_PROVIDER)) {
+                    params = new GoogleBuilder().setScopes(permissions).build().getParams();
+                } else if (mProviderId.equals(FACEBOOK_PROVIDER)) {
+                    params = new FacebookBuilder().setPermissions(permissions).build().getParams();
+                } else {
+                    params = new Bundle();
+                }
+                setParams(params);
                 return this;
             }
 
             @NonNull
+            protected Bundle getParams() {
+                return mParams;
+            }
+
+            @NonNull
+            @Deprecated
             public Builder setParams(@Nullable Bundle params) {
-                mParams = params;
+                mParams.putAll(params == null ? new Bundle() : params);
+                return this;
+            }
+
+            @CallSuper
+            @NonNull
+            public IdpConfig build() {
+                return new IdpConfig(mProviderId, mParams);
+            }
+        }
+
+        public static final class EmailBuilder extends Builder {
```

Comment:
Public classes should have JavaDoc.

Reply:
Yep, see https://github.com/firebase/FirebaseUI-Android/pull/1083#discussion_r159293138.